cocotb/utils.py

```diff
@@ -202,97 +202,127 @@ def unpack(ctypes_obj, string, bytes=None):
 import cocotb.ANSI as ANSI
 
 
-def _sane_color(x):
+# A note on the use of latin1 in the deprecations below:
+# Latin1 is the only encoding `e` that satisfies
+# `all(chr(x).encode(e) == bytes([x]) for x in range(255))`
+# Our use of `ord` and `chr` throughout other bits of code make this the most
+# compatible choice of encoding. Under this convention, old code can be upgraded
+# by changing "binary" strings from `"\x12\x34"` to `b"\x12\x34"`.
```

Comment:
Maybe that "upgrade information" would be nice to have in the deprecation warnings.

Reply:
This upgrade information isn't particularly applicable to callers of `hexdump`, who will not notice the difference in behavior. Users will be dumping data they got from their simulation, not constants from their python.